specification/redis/resource-manager/Microsoft.Cache/preview/2023-05-01/redis.json

```diff
@@ -1850,6 +1850,536 @@
           }
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Cache/redis/{cacheName}/accessPolicies/{accessPolicyName}": {
+      "put": {
+        "tags": [
+          "Redis",
+          "AAD"
+        ],
+        "operationId": "AccessPolicy_CreateUpdate",
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "azure-async-operation"
+        },
+        "x-ms-examples": {
+          "RedisCacheAccessPolicyCreateUpdate": {
+            "$ref": "./examples/RedisCacheAccessPolicyCreateUpdate.json"
+          }
+        },
+        "description": "Adds an access policy to the redis cache",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "cacheName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the Redis cache.",
+            "pattern": ".*"
+          },
+          {
+            "name": "accessPolicyName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the access policy that is being added to the Redis cache.",
+            "pattern": "^[a-zA-Z0-9]+$"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/RedisCacheAccessPolicyCreateUpdateParameters"
```

Comment:
>RedisCacheAccessPolicyCreateUpdateParameters [](http://example.com/codeflow?start=37&length=44)

RE linter warning: A PUT operation request body schema should be the same as its 200 response schema, to allow reusing the same entity between GET and PUT. If the schema of the PUT request body is a superset of the GET response body, make sure you have a PATCH operation to make the resource updatable. Operation: 'AccessPolicy_CreateUpdate' Request Model: 'parameters[3].schema' Response Model: 'responses[200].schema'
Location: Microsoft.Cache/preview/2023-05-01/redis.json#L1855

You should just define one RedisCacheAccessPolicy definition used for both PUT requests and GET responses. non-puttable fields can be marked as readonly, or with x-ms-mutability

Reply:
Using common definition for request and response now